File: cpp_packages/cpp_pkg/src/resistance_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "robot_interface/srv/resistance_system.hpp"

class ResistanceServer: public rclcpp::Node{

    private:

        rclcpp::Service<robot_interface::srv::ResistanceSystem>::SharedPtr server_;

        void resistance_callback(robot_interface::srv::ResistanceSystem::Request::SharedPtr request, 
                                 robot_interface::srv::ResistanceSystem::Response::SharedPtr response)
        {
            response->resistance = request->tension/request->current;
            float resistance = response->resistance;

            RCLCPP_INFO(this->get_logger(), "Resistance: %.2f", resistance);
        }

    public:

        ResistanceServer():Node("resistance_server"){

            server_ = this->create_service<robot_interface::srv::ResistanceSystem>("resistance_measure",
                                            std::bind(&ResistanceServer::resistance_callback, this,
                                            std::placeholders::_1, std::placeholders::_2));
            RCLCPP_INFO(this->get_logger(), "Resistance Server has been initialized");
        }

};


int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<ResistanceServer>();
    rclcpp::spin(node);
    rclcpp::shutdown();

}

File: python_packages/python_pkg/python_pkg/resistance_server.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from robot_interface.srv import ResistanceSystem

class ResistanceServer(Node):
    def __init__(self):
        super().__init__("resistance_server")
        self.server_ = self.create_service(ResistanceSystem, "resistance_measure", 
                                           self.resistance_callback)
        self.get_logger().info("Resistance server has been initialized")
    

    def resistance_callback(self, request, response):
        response.resistance = request.tension/request.current

        self.get_logger().info(f"Resistance: {response.resistance}")

        return response


def main(args = None):
    rclpy.init(args=args)
    node = ResistanceServer()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

